app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.weatherapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.work.*
+import com.example.weatherapp.databinding.ActivityMainBinding
+import com.example.weatherapp.workers.RequestWeatherWorker
+
+class MainActivity : AppCompatActivity() {
+    private var lat = 53.43
+    private var lon = 14.55
+    private var API_KEY = "680f2f4fd0536de70ad6a0f3bcd81790"
+    private var binding: ActivityMainBinding? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding?.root)
+
+        binding?.buttonGetWeather?.setOnClickListener {
+            buttonGetWeatherListener()
+        }
+    }
+
+    private fun buttonGetWeatherListener() {
+        val constraints: Constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+        val data: Data = Data.Builder()
+            .putString("API_KEY", API_KEY)
+            .putDouble("lon", lon)
+            .putDouble("lat", lat)
+            .build()
+        val workRequest: WorkRequest = OneTimeWorkRequest.Builder(RequestWeatherWorker::class.java)
+            .setInputData(data)
+            .setConstraints(constraints)
+            .build()
+        WorkManager.getInstance(applicationContext).enqueue(workRequest)
+        WorkManager.getInstance(applicationContext)
+            .getWorkInfoByIdLiveData(workRequest.id)
+            .observe(this@MainActivity, Observer { workInfo: WorkInfo ->
+                when (workInfo.state) {
+                    WorkInfo.State.SUCCEEDED -> {
+                        val weatherString = buildString {
+                            append("\nweatherMain: ")
+                            append(workInfo.outputData.getString("weatherMain"))
+                            append("\nweatherDescription: ")
+                            append(workInfo.outputData.getString("weatherDescription"))
+                            append("\ntemp: ")
+                            append(workInfo.outputData.getDouble("temp", 0.0))
+                            append("\nfeelsLike: ")
+                            append(workInfo.outputData.getDouble("feelsLike", 0.0))
+                            append("\ntempMin: ")
+                            append(workInfo.outputData.getDouble("tempMin", 0.0))
+                            append("\ntempMax: ")
+                            append(workInfo.outputData.getDouble("tempMax", 0.0))
+                        }
+                        binding?.textWeather?.text = weatherString
```

Comment:
I suggest separating different data into different textViews so we can make UI nicer

Reply:
@kozych I agree with above.
Also please display information with current date and time so we can see that after clicking the button the data is refreshed from the Internet.

Reply:
done
also added "last update" info which stores time and date of last successful request for weather. Im also storing these infos in sharedPreferences